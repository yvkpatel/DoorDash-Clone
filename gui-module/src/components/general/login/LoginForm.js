/**
 * This is the form on the login page that users can submit their login detial
 *
 * @param: onLogin -> function that handles the submission of form data
 */
import { React, useState } from "react";
import Button from "react-bootstrap/Button";
import { Form } from "react-bootstrap";
import "./Login.css";

const LoginForm = ({ onLogin }) => {
  const [email, setEmail] = useState("");
  const [password, setPassword] = useState("");
  const [userType, setUserType] = useState("");

  const onSubmit = (e) => {
    e.preventDefault();
    if (!email || !password || !userType) {
      alert("Please input all details");
      return;
    }
    //console.log({ email });
    onLogin({ email, password, userType });
  };

  return (
    <div data-testid="loginForm">
      <h3 className="login-text">
        <strong>SIGN IN</strong>
      </h3>
      <Form onSubmit={onSubmit}>
        <Form.Group className="details" data-testid="textfields">
          <Form.Label>Username</Form.Label>
          <Form.Control
            type="text"
            placeholder="pizza@gmail.com"
            value={email}
            data-testid="emailfield"
            onChange={(e) => setEmail(e.target.value)}
          />
        </Form.Group>

        <Form.Group className="details">
          <Form.Label>Password</Form.Label>
          <Form.Control
            type="password"
            placeholder="Password"
            value={password}
            data-testid="passwordfield"
            onChange={(e) => setPassword(e.target.value)}
          />
        </Form.Group>

        <Form.Group className="details" data-testid="signinfield">
          <Form.Label>Sign in as:</Form.Label>
          {/* <Form.Control type="user type" placeholder="User Type" value={userType} onChange={(e) => setUserType(e.target.value)} /> */}
          <Form.Select
            aria-label="Default select example"
            data-testid="select"
            type="user type"
            placeholder="User Type"
            value={userType}
            onChange={(e) => setUserType(e.target.value)}
          >
            <option>Choose User Type</option>
            <option value="Customer">Customer</option>
            <option value="Driver">Driver</option>
            <option value="Restaurant">Restaurant</option>
          </Form.Select>
        </Form.Group>
        <Button
          variant="primary"
          type="submit"
          className="btn-primary"
          data-testid="goButton"
        >
          {" "}
          Go{" "}
        </Button>
      </Form>
    </div>
  );
};

export default LoginForm;

//test selection of drop down
