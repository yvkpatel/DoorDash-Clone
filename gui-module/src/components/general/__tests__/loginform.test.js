import {
  render,
  screen,
  cleanup,
  fireEvent,
  waitFor,
} from "@testing-library/react";
import renderer from "react-test-renderer";
import LoginForm from "../login/LoginForm";
import userEvent from "@testing-library/user-event";
import { Form } from "react-bootstrap";
import "@testing-library/jest-dom/extend-expect";

afterEach(() => {
  cleanup();
});

test("should render LoginForm component", () => {
  render(<LoginForm />);
  const loginformElement = screen.getByTestId("loginForm");
  expect(loginformElement).toBeInTheDocument();
});

test("should show text fields", () => {
  render(<LoginForm />);
  const textElement = screen.getByTestId("textfields");
  expect(textElement).toBeInTheDocument();
});

test("should show Go/login button", () => {
  render(<LoginForm />);
  const button = screen.getByTestId("goButton");
  expect(button).toBeInTheDocument();
});

test("selecting an option is working", () => {
  const { getByRole, getByText } = render(<LoginForm />);
  const controlElement = getByText("Customer");
  fireEvent.click(controlElement);
  fireEvent.click(getByText(controlElement.value));
  waitFor(() => expect(controlElement).toHaveTextContent(controlElement.value));
});

test("matches snapshot", () => {
  const tree = renderer.create(<LoginForm />);
  expect(tree).toMatchSnapshot();
});
