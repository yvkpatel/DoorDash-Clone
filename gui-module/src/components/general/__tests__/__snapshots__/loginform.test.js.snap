// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  data-testid="loginForm"
>
  <h3
    className="login-text"
  >
    <strong>
      SIGN IN
    </strong>
  </h3>
  <form
    className=""
    onSubmit={[Function]}
  >
    <div
      className="details"
      data-testid="textfields"
    >
      <label
        className="form-label"
      >
        Username
      </label>
      <input
        className="form-control"
        data-testid="emailfield"
        onChange={[Function]}
        placeholder="pizza@gmail.com"
        type="text"
        value=""
      />
    </div>
    <div
      className="details"
    >
      <label
        className="form-label"
      >
        Password
      </label>
      <input
        className="form-control"
        data-testid="passwordfield"
        onChange={[Function]}
        placeholder="Password"
        type="password"
        value=""
      />
    </div>
    <div
      className="details"
      data-testid="signinfield"
    >
      <label
        className="form-label"
      >
        Sign in as:
      </label>
      <select
        aria-label="Default select example"
        className="form-select"
        data-testid="select"
        onChange={[Function]}
        placeholder="User Type"
        type="user type"
        value=""
      >
        <option>
          Choose User Type
        </option>
        <option
          value="Customer"
        >
          Customer
        </option>
        <option
          value="Driver"
        >
          Driver
        </option>
        <option
          value="Restaurant"
        >
          Restaurant
        </option>
      </select>
    </div>
    <button
      className="btn-primary btn btn-primary"
      data-testid="goButton"
      disabled={false}
      type="submit"
    >
       
      Go
       
    </button>
  </form>
</div>
`;
